packages/story-editor/src/elements/video/trim.js

```diff
@@ -89,17 +126,54 @@ function VideoTrim({ box, element }) {
     [setVideoNode]
   );
 
+  const fetchOriginalResource = useCallback(() => {
+    getMediaById(original)
+      .then((_originalResource) => {
+        setIsLoading(false);
+        setOriginalResource(_originalResource);
+        setOriginalEndOffset(getMsFromHMS(end));
+        setOriginalStartOffset(getMsFromHMS(start));
+      })
+      .catch(() => {
+        setOriginalResource(false);
+        setIsLoading(false);
+      });
+  }, [
+    original,
+    getMediaById,
+    setOriginalResource,
+    start,
+    end,
+    setOriginalEndOffset,
+    setOriginalStartOffset,
+    getMsFromHMS,
+  ]);
```

Comment:
I know this is only WIP, but I'd really like to see this moved to the provider instead if possible. Maybe as a third custom hook there or alternatively built into the `useVideoNode` hook.

Reply:
@barklund Thanks for checking. 

Some refactoring has been done now -- if you have time, please review the PR to see if you notice any other general places for improvement. There are still a few outstanding things but generally it should be getting ready-ish.